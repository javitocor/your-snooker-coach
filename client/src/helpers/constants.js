export const DEV_URL = 'https://localhost:3000/api/v1';

export const GET_ALL_PLAYERS = 'GET_ALL_PLAYERS';
export const GET_ALL_PLAYERS_ERROR = 'GET__PLAYERS_ERROR';
export const GET_ALL_PLAYERS_PENDING = 'GET_ALL_PLAYERS_PENDING';
export const GET_PLAYER = 'GET_PLAYER';
export const GET_PLAYER_ERROR = 'GET_PLAYER_ERROR';
export const GET_PLAYER_PENDING = 'GET_PLAYER_PENDING';


export const initialStatePlayers = {
  error: null,
  pending: false,
  playersList: [],
  player: [],
}

export const GET_TRAININGS = 'GET_TRAININGS';
export const GET_TRAININGS_PENDING = 'GET_TRAININGS_PENDING';
export const GET_TRAININGS_ERROR = 'GET_TRAININGS_ERROR';
export const CREATE_TRAINING = 'CREATE_TRAINING';
export const CREATE_TRAINING_PENDING = 'CREATE_TRAINING_PENDING';
export const CREATE_TRAINING_ERROR = 'CREATE_TRAINING_ERROR';

