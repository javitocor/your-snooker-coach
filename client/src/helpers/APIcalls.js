import { DEV_URL } from './constants';

export const AllPlayersCall = () => async dispatch => {
  const Url = DEV_URL;
  try {
      dispatch(getAllPlayersPending());

    const response = await fetch(`${Url}/players`, { mode: 'cors' });
    const playersList = await response.json();
    dispatch(getAllPlayers(playersList));
    return playersList;
  } catch (error) {
      dispatch(getAllPlayersError(error));
  }
};

export const PlayerCall = (id) => async dispatch => {
  const Url = DEV_URL;
  try {
      dispatch(getPlayerPending());

    const response = await fetch(`${Url}/players/${id}`, { mode: 'cors' });
    const player = await response.json();
    dispatch(getPlayer(player));
    return player;
  } catch (error) {
      dispatch(getPlayerError(error));
  }
};