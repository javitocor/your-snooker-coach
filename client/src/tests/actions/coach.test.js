import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import { PlayerCall } from '../../helpers/APIcalls';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('async actions', () => {
  let store;
  const player = [
    {
      name: 'Mark J Williams',
      nationality: 'Wales',
      born: '1975-03-21',
      firstseason: '1992',
      info: `Ranking titles: 22
      Triple Crown wins: Seven (Three World Championship, two UK Championships, two Masters)
      
      Career highlights
      
      1996: Wins his first two ranking titles, at the Welsh Open and Grand Prix
      1998: Beats Stephen Hendry 10-9 in the famous re-spotted black Masters final at Wembley
      1999: Wins the UK Championship for the first time
      2000: Wins his first World Championship title, coming from 13-7 down to beat Matthew Stevens 18-16 in an all-Welsh final
      2003: Wins second world title, beating Ken Doherty 18-16 in the Crucible final.  Becomes the third player, after Steve Davis and Stephen Hendry, to win the World, UK and Masters titles in the same season.
      2005: Makes his first 147 at the Crucible, worth a handy Â£161,000
      2011: Wins the 18th world ranking title of his career at the German Masters
      2017: Secures his first ranking event victory in six years at the Northern Ireland Open, beating Yan Bingtao 9-8 in the final
      2018: Wins the German Masters. Wins the World Championship for the third time, beating John Higgins 18-16 in one of the all-time great Crucible  finals. At the age of 43, becomes the oldest winner since Ray Reardon in 1978. Joins Mark Selby and John Spencer on three world titles. Wins the Yushan World Open in China, his 22nd career ranking title.
      
      `,
      image: 'http://snooker.org/img/players/MarkWilliams.png',
      ranking: 10,
      rate: '$ 154/hr',
    },
  ];
  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(() => {
    nock.cleanAll();
  });

  it('creates getPlayer action', () => {
    nock('https://localhost:3000')
      .get('/api/v1/player/1')
      .reply(200, player);

    return store.dispatch(PlayerCall(1))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
