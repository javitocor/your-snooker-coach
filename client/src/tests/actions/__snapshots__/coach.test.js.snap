// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions creates getPlayer action 1`] = `
Array [
  Object {
    "type": "GET_PLAYER_PENDING",
  },
  Object {
    "error": [TypeError: Network request failed],
    "type": "GET_PLAYER_ERROR",
  },
]
`;
